src/configUrl.ts

```diff
@@ -1,30 +1,9 @@
-/**
- * determines the config URL based on environment
- */
-
-export function getKumaCpServerUrl(): string {
-  // Uses an absolute URL during testing
-  // because the Node fetch polyfill we’re using requires it.
-  if (process.env.NODE_ENV === 'test') {
-    return 'http://localhost/'
-  }
-
-  if (process.env.NODE_ENV === 'development') {
-    return process.env.VUE_APP_KUMA_CONFIG?.replace('/config', '/') || ''
-  } else {
-    const href = window.location.href
-
-    return `${href.substring(0, href.indexOf('/gui'))}/`
-  }
-}
-
 export function kumaDpServerUrl(): string {
-  const url = window.location
-  const envConfig = String(process.env.VUE_APP_KUMA_DP_SERVER_URL)
+  const dpServerUrl = String(import.meta.env.VITE_KUMA_DP_SERVER_URL)
 
-  if (process.env.NODE_ENV === 'development') {
-    return envConfig
+  if (import.meta.env.PROD) {
+    return dpServerUrl.replace('localhost', window.location.hostname)
```

Comment:
I think I'm fine with this but in general I'm not really sure 😄. I think I'm most used to a `frontend` having just an env variable called `API_URL` that points to the API. In the future they might not be hosted on the same domain.

Reply:
Previously, we had `VUE_APP_KUMA_DP_SERVER_URL=https://localhost:5678/` set in `.env`.

In essence, this function becomes:

```js
function kumaDpServerUrl(): string {
  const url = window.location
  const envConfig = 'https://localhost:5678/'

  if (process.env.NODE_ENV === 'development') {
    return envConfig
  }

  return envConfig.replace('localhost', url.hostname)
}
```

Simplifying this:

```js
function kumaDpServerUrl(): string {
  if (process.env.NODE_ENV === 'development') {
    return 'https://localhost:5678/'
  } else {
    return 'https://localhost:5678/'.replace('localhost', window.location.hostname)
  }
}
```

and further:

```js
function kumaDpServerUrl(): string {
  if (process.env.NODE_ENV === 'development') {
    return 'https://localhost:5678/'
  } else {
    return `https://${window.location.hostname}:5678/`
  }
}
```

So yes, the concern of this not running on the same domain is valid: If it does run from a different domain, requests to this URL would fail.

However, this is already not configurable. You would need to rebuild the frontend application with a different environment variable value and that’s not something a user of Kuma can do. To make this a user choice, this would need to come from the API so we can make it dynamic (probably from `/` or `/config` [I wish the APIs were prefixed with `/api`]).